mtags-shared/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala

```diff
@@ -77,6 +79,9 @@ abstract class CompilerAccess[Reporter, Compiler](
    * Asynchronously execute a function on the compiler thread with `Thread.interrupt()` cancellation.
    */
   def withInterruptableCompiler[T](
+      actionName: String,
+      actionInfo: () => String
```

Comment:
We could pass `VirtualFileParams` and match on a particular subtype to only do it once here.